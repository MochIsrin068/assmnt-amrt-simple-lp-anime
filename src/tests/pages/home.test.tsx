import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import HomePage from "../../pages/home";

describe("Perform testing page home", () => {
  // Handle error : TypeError: window.matchMedia is not a function
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      value: jest.fn(() => ({
        matches: true,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      })),
    });
  });

  beforeEach(() => {
    global.console.error = (...args) => {
      const propTypeFailures = [/Error:/];
      if (propTypeFailures.some((prop) => prop.test(args[0]))) {
      }
    };
  });

  it("Perform snapshots test HomePage", () => {
    const view = render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<HomePage />} />
        </Routes>
      </BrowserRouter>
    );
    expect(view).toMatchSnapshot();
  });

  it("test title header", () => {
    render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<HomePage />} />
        </Routes>
      </BrowserRouter>
    );
    const label = screen.getAllByText(/The Anime Movie/i);
    expect(label[0]).toBeInTheDocument();
  });

  it("test description header", () => {
    render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<HomePage />} />
        </Routes>
      </BrowserRouter>
    );
    const label = screen.getAllByText(/The Anime Movie/i);
    expect(label[1]).toBeInTheDocument();
  });
});
