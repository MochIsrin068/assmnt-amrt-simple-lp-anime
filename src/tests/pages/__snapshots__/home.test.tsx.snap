// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Perform testing page home Perform snapshots test HomePage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="home"
      >
        <header
          class="header --header-home"
        >
          <div
            class="header__jumbotron"
          >
            <h1>
              The Anime Movie
            </h1>
            <p>
              The Anime Movie is a web-based application to see what movies are trending and popular that you can use as a reference to watch with your special someone :)
            </p>
          </div>
        </header>
        ;
        <div
          class="input-search"
        >
          <input
            placeholder="Search anime movie..."
            type="text"
          />
        </div>
        <div
          class="home__content"
        >
          <div
            class="home__content__list"
          >
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
            <div
              class="loader --list"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 200px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <div>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="home"
    >
      <header
        class="header --header-home"
      >
        <div
          class="header__jumbotron"
        >
          <h1>
            The Anime Movie
          </h1>
          <p>
            The Anime Movie is a web-based application to see what movies are trending and popular that you can use as a reference to watch with your special someone :)
          </p>
        </div>
      </header>
      ;
      <div
        class="input-search"
      >
        <input
          placeholder="Search anime movie..."
          type="text"
        />
      </div>
      <div
        class="home__content"
      >
        <div
          class="home__content__list"
        >
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
          <div
            class="loader --list"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton"
                style="height: 200px;"
              >
                ‌
              </span>
              <br />
            </span>
            <div>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
                <span
                  class="react-loading-skeleton"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
